test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/TagHelperParseTreeRewriterTest.cs

```diff
@@ -1108,6 +1108,91 @@ public static TheoryData AllowedChildrenData
             EvaluateData(descriptors, documentContent, (MarkupBlock)expectedOutput, (RazorDiagnostic[])expectedErrors);
         }
 
+        [Fact]
+        public void Rewrite_AllowsHtmlCommentsAsChildren()
+        {
+            // Arrangestring documentContent,
+            IEnumerable<string> allowedChildren = new List<string> { "b" };
+            string literal = "asdf";
+            string commentOutput = $"<!--Hello World-->";
```

Comment:
Given that this is an integration test I'd make this a little more complex: `<!-- Hello @world -->`.

Also, no need for `$`